### PR TITLE
[#10] Add peer roles config

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Comment by fqutishat** [comment]
verify

**Comment by fqutishat** [comment]
verify

**Comment by fqutishat** [comment]
verify

File: protos/gossip/message.proto [hunk 1]
```diff
@@ -130,6 +130,12 @@ message GossipMessage {
         // Encapsulates private data used to distribute
         // private rwset after the endorsement
         PrivateDataMessage private_data = 25;
+
+        // Contains validation results
+        ValidationResultsMessage validation_results_msg = 50;
```
Comment (troyronda) [inline protos/gossip/message.proto]: i wonder if we should leave a larger gap.

File: extensions/roles/ledger_roles_config.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+// IsValidator returns true if the peer is a validator
+func IsValidator() bool {
+	return true
+}
+
+// RolesAsString returns the roles for the peer
+func RolesAsString() []string {
+	return []string{}
+}
```
Comment (troyronda) [inline extensions/roles/ledger_roles_config.go]: missing new line at end of file

File: extensions/roles/ledger_roles_config.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
```
Comment (troyronda) [inline extensions/roles/ledger_roles_config.go]: are there tests coming?

File: protos/gossip/message.proto (hunk 1 above)
Comment (fqutishat) [reply protos/gossip/message.proto]: We will increase it in the next change

File: extensions/roles/ledger_roles_config.go (hunk 3 above)
Comment (fqutishat) [reply extensions/roles/ledger_roles_config.go]: next change

File: extensions/roles/ledger_roles_config.go (hunk 2 above)
Comment (fqutishat) [reply extensions/roles/ledger_roles_config.go]: i will add it

File: extensions/roles/ledger_roles_config.go (hunk 3 above)
Comment (fqutishat) [reply extensions/roles/ledger_roles_config.go]: done

File: extensions/roles/ledger_roles_config.go (hunk 2 above)
Comment (fqutishat) [reply extensions/roles/ledger_roles_config.go]: done

**Comment by fqutishat** [comment]
verify

**Comment by fqutishat** [comment]
verify

File: extensions/roles/ledger_roles_config_test.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+package roles
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestIsCommitter(t *testing.T) {
+	require := require.New(t)
```
Comment (llorllale) [inline extensions/roles/ledger_roles_config_test.go]: @fqutishat you can just do `require.True(t, IsCommitter())` directly - no need to instantiate `require` in all these tests

Comment (fqutishat) [reply extensions/roles/ledger_roles_config_test.go]: done

File: protos/gossip/message.proto (hunk 1 above)
Comment (bstasyszyn) [inline protos/gossip/message.proto]: This doesn't belong here - it will be part of a future commit (i.e. distributed validation).

File: protos/gossip/message.proto [hunk 5]
```diff
@@ -130,6 +130,12 @@ message GossipMessage {
...
+
+        // Request for block validaton
+        DataMessage validation_req_msg = 51;
```
Comment (bstasyszyn) [inline protos/gossip/message.proto]: Same - should not be part of this commit.

File: protos/gossip/message.proto [hunk 6]
```diff
@@ -369,4 +376,10 @@ message Chaincode {
     string name = 1;
     string version = 2;
     bytes metadata = 3;
-}
\ No newline at end of file
+}
+
+// ValidationResultsMessage is the message containing block validation results
+message ValidationResultsMessage {
```
Comment (bstasyszyn) [inline protos/gossip/message.proto]: Same - not part of this commit.

Comment (fqutishat) [reply protos/gossip/message.proto]: done

File: protos/gossip/message.proto (hunk 5 above)
Comment (fqutishat) [reply protos/gossip/message.proto]: done

File: protos/gossip/message.proto (hunk 1 above)
Comment (fqutishat) [reply protos/gossip/message.proto]: done

Comment (fqutishat) [reply protos/gossip/message.proto]: I remove it

**Comment by fqutishat** [comment]
verify

**Comment by troyronda** [comment]
LGTM
